File: TypeScript/src/Entities/Person.ts
import { Entity } from "./Entity";
import { CanAdmire } from "./CanAdmire";

export class Person extends Entity implements CanAdmire {
    constructor(name: string) {
        super(name);
        this.setNoise("Wow, that's so beautiful");
    }

    public admire(admiring: Entity): string {
        return `${this.getNoise()} at ${admiring.getName()}`;
    }
}

File: src/main/java/csci/ooad/grad/Entities/Person.java
package csci.ooad.grad.Entities;

public class Person extends Entity implements CanAdmire {

    public Person(String name) {
        super(name);

        setNoise("Wow, that's so beautiful");
    }

    public String admire(Entity admiring) {
        return getNoise() + " at " + admiring.getName();
    }
}